micromasters_import/management/commands/queries/002_courserun_enrollment.sql

```diff
@@ -0,0 +1,65 @@
+--courserun migration from MicroMaster to MITxOnline
+INSERT INTO public.courses_courserun(
+    title,
+    courseware_id,
+    run_tag,
+    courseware_url_path,
+    start_date,
+    end_date,
+    enrollment_start,
+    enrollment_end,
+    expiration_date,
+    live,
+    created_on,
+    updated_on,
+    course_id)
+SELECT
+    mm_courserun.title,
+    mm_courserun.edx_course_key,
+    LEFT(CASE WHEN position('+' in mm_courserun.edx_course_key) > 0
+    THEN (regexp_match(mm_courserun.edx_course_key, '[^+]+$'))[1]
+    ELSE (regexp_match(mm_courserun.edx_course_key, '[^/]+$'))[1] END, 10) AS run_tag,
+    mm_courserun.enrollment_url,
+    mm_courserun.start_date,
+    mm_courserun.end_date,
+    mm_courserun.enrollment_start,
+    mm_courserun.enrollment_end,
+    mm_courserun.enrollment_end + INTERVAL '1 day',
+    NOT mm_courserun.is_discontinued,
+    coalesce(mm_courserun.enrollment_start, now()), --- need to confirm the default if enrollment_start is null
+    coalesce(mm_courserun.enrollment_end, now()),
+    pk_map.course_id
+FROM micromasters.courses_courserun AS mm_courserun
+JOIN public.micromasters_import_courseid AS pk_map ON mm_courserun.course_id = pk_map.micromasters_id
+LEFT JOIN public.courses_courserun AS mo_courserun ON mo_courserun.courseware_id = mm_courserun.edx_course_key
+WHERE mo_courserun.courseware_id IS NULL ---bring in course runs that don't exist in MITxOnline
+ON CONFLICT DO NOTHING; -- Do we want to update?
+
+--courserun Enrollment migration from MicroMaster to MITxOnline
+INSERT INTO public.courses_courserunenrollment(
+    run_id,
+    user_id,
+    change_status,
+    active,
+    edx_enrolled,
+    edx_emails_subscription,
+    enrollment_mode,
+    created_on,
+    updated_on)
+SELECT
+    mo_courserun.id,
+    mo_user.id,
+    '' as change_status,
+    (mm_enrollment.data->>'is_active')::boolean,
+    TRUE,
+    FALSE,
+    mm_enrollment.data->>'mode',
+    ((mm_enrollment.data->>'created')::timestamp), -- need to verify
+    ((mm_enrollment.data->>'created')::timestamp)
```

Comment:
created, or updated?

Reply:
@pdpinch @rhysyngsun Looks like there isn't updated field in McroMaster enrollment table
There are also some enrollments on RC that have blank 'created'. For those cases, how should we populate created_on and updated_on ?
![image](https://user-images.githubusercontent.com/3138890/182212358-917e24f8-81e6-4938-86d3-11f73fd7b604.png)
Also, just to confirm that we are not filtering out any enrollments as long as mm_social.provider = 'mitxonline'?

Reply:
I think we should filter the enrollments just down to DEDP ones (we'll also need to do this to orders). My thinking is those other programs only use edx.org and I think there could be a lot of learner confusion seeing enrollments over in mitxonline that may not be current and not integrated with edx.org. @pdpinch do you agree?

As far as the two fields, we might just have to use `enrollment_start` and `enrollment_end` as best guesses, similar to my comment about course runs.

Reply:
@rhysyngsun My thinking is that the new table 'micromasters_import_courseid' will only be DEDP courses mapping if we just want to bring in DEDP courserun and enrollment for now, so we don't have to add any hardcoded filter here. Also the JOIN public.courses_courserun here should be enough to limit the enrollments down to DEDP. If we want to be sure, I would add a filter to courserun query

Reply:
I agree with the goal of only importing DEDP enrollments. I won't comment on the implementation.